desktop/src/client/proxy/manual-proxy-configuration-window/manual-proxy-configuration-window.html

```diff
@@ -5,21 +5,24 @@
     <link href="./manual-proxy-configuration-window.css" rel="stylesheet" type="text/css">
 </head>
 <script>
-    const { ipcRenderer, remote } = require("electron");
-    const initial = remote.getCurrentWindow().currentSettings;
+    const { ipcRenderer } = require("electron");
 
-    document.addEventListener('DOMContentLoaded', function () {
+    const configDeferred = ipcRenderer.invoke("proxy-configuration-requested");
+
+    document.addEventListener("DOMContentLoaded", function () {
         window.urlEl = document.getElementById("url");
         window.portEl = document.getElementById("port");
         window.usernameEl = document.getElementById("username");
         window.passwordEl = document.getElementById("password");
 
-        if (initial) {
-            urlEl.value = initial.url || "";
-            portEl.value = initial.port || "";
-            usernameEl.value = initial.username || "";
-            passwordEl.value = initial.password || "";
-        }
+        configDeferred.then((settings) => {
+            if (settings) {
+                window.urlEl.value = settings.url || "";
```

Comment:
Should we block user interaction or keep the window hidden until this promise resolves? It seems like there could be a race condition here if the window loads, the user starts typing, then the promise resolves and overwrites the values.